pkg/ddc/juicefs/operations/base.go

```diff
@@ -0,0 +1,301 @@
+	}
+	return
+}
+
+// The count of the JuiceFS Filesystem
+func (j JuiceFileUtils) Count(juiceSubPath string) (total int64, err error) {
+	var (
+		command = []string{"du", "-sb", juiceSubPath}
+		stdout  string
+		stderr  string
+		utotal  uint64
+	)
+
+	stdout, stderr, err = j.exec(command, false)
+	if err != nil {
+		err = fmt.Errorf("execute command %v with expectedErr: %v stdout %s and stderr %s", command, err, stdout, stderr)
+		return
+	}
+
+	// [File Count Folder Count Total Bytes 1152 4 154262709011]
+	str := strings.Split(stdout, "\n")
+
+	if len(str) != 1 {
+		err = fmt.Errorf("failed to parse %s in Count method", str)
+		return
+	}
+
+	data := strings.Fields(str[0])
+	if len(data) != 2 {
+		err = fmt.Errorf("failed to parse %s in Count method", data)
+		return
+	}
+
+	utotal, err = strconv.ParseUint(data[0], 10, 64)
+	if err != nil {
+		return
+	}
+
+	return int64(utotal), err
+}
+
+// file count of the JuiceFS Filesystem (except folder)
+// use "ls -lR  xxx|grep "^-"| wc -l"
+func (j JuiceFileUtils) GetFileCount(juiceSubPath string) (fileCount int64, err error) {
+	var (
+		//strs    = "du -ah juiceSubPath |grep ^- |wc -l "
+		strs    = fmt.Sprintf("du -ah %s |grep ^- |wc -l ", juiceSubPath)
+		command = []string{"bash", "-c", strs}
+		stdout  string
+		stderr  string
+	)
+
+	stdout, stderr, err = j.exec(command, false)
+	if err != nil {
+		err = fmt.Errorf("execute command %v with expectedErr: %v stdout %s and stderr %s", command, err, stdout, stderr)
+		return
+	}
+
+	// eg: Master.FilesCompleted  (Type: COUNTER, Value: 6,367,897)
+	str := strings.Split(stdout, "\n")
+
+	if len(str) != 1 {
+		err = fmt.Errorf("failed to parse %s in Count method", str)
+		return
+	}
+
+	data := strings.Fields(str[0])
+	if len(data) != 1 {
+		err = fmt.Errorf("failed to parse %s in Count method", data)
+		return
+	}
+
+	fileCount, err = strconv.ParseInt(data[0], 10, 64)
+	if err != nil {
+		return
+	}
+
+	return fileCount, nil
+}
+
+// Mkdir mkdir in juicefs container
+func (j JuiceFileUtils) Mkdir(juiceSubPath string) (err error) {
+	var (
+		command = []string{"mkdir", juiceSubPath}
+		stdout  string
+		stderr  string
+	)
+
+	stdout, stderr, err = j.exec(command, true)
+	if err != nil {
+		if strings.Contains(stdout, "File exists") {
+			err = nil
+		} else {
+			err = fmt.Errorf("execute command %v with expectedErr: %v stdout %s and stderr %s", command, err, stdout, stderr)
+			return
+		}
+	}
+	return
+}
+
+// DeleteDir delete dir in pod
+func (j JuiceFileUtils) DeleteDir(dir string) (err error) {
+	var (
+		command = []string{"rm", "-rf", dir}
+		stdout  string
+		stderr  string
+	)
+
+	stdout, stderr, err = j.exec(command, true)
+	if err != nil {
+		err = fmt.Errorf("execute command %v with expectedErr: %v stdout %s and stderr %s", command, err, stdout, stderr)
+		return
+	}
+	return
+}
+
+// GetMetric Get pod metrics
+func (j JuiceFileUtils) GetMetric() (metrics string, err error) {
+	var (
+		command = []string{"curl", "0.0.0.0:9567/metrics"}
```

Comment:
Is this 9567 port configurable in JuiceFS?

Reply:
It's configurable, it will be set configurable next step.